src/turret.cpp

```diff
@@ -589,16 +590,20 @@ int vehicle::automatic_fire_turret( vehicle_part &pt )
         if( auto_target == nullptr ) {
             if( boo_hoo ) {
                 cpu.name = string_format( pgettext( "vehicle turret", "The %s" ), pt.name() );
-                if( u_see ) {
+                // check if the player can see or hear then print chooses a message accordingly
+                if( u_see & u_hear ) {
                     add_msg( m_warning, ngettext( "%s points in your direction and emits an IFF warning beep.",
                                                   "%s points in your direction and emits %d annoyed sounding beeps.",
                                                   boo_hoo ),
                              cpu.name, boo_hoo );
-                } else {
+                } else if( !u_see & u_hear ) {
                     add_msg( m_warning, ngettext( "%s emits an IFF warning beep.",
                                                   "%s emits %d annoyed sounding beeps.",
                                                   boo_hoo ),
                              cpu.name, boo_hoo );
```

Comment:
I think we should rewrite the text for this section. In this case we can't see what's beeping, right? So the text should be something like `You hear a beep.` and `You hear %d annoyed sounding beeps.`

Reply:
I made some changes based on your feedback please take a look when you have some time.
sorry for taking too much time and making too many commits on something that is supposed to be simple, this is my first contribution in a large project and it appears I got a little too confident.

Reply:
Looks good to me. Good work.
Even a simple change could take a lot of time, especially for first-time contributor, so no need to apologize.